juju/model/__init__.py

```diff
@@ -3064,16 +3103,21 @@ async def wait_for_idle(
             wait_for_at_least_units if wait_for_at_least_units is not None else 1
         )
 
-        timeout = timedelta(seconds=timeout) if timeout is not None else None
-        idle_period = timedelta(seconds=idle_period)
+        timeout_ = timedelta(seconds=timeout) if timeout is not None else None
+        idle_period_ = timedelta(seconds=idle_period)
```

Comment:
Names like this always strike me as potentially very error prone for future changes. Is the rename to get the types right?

Reply:
yes, and I explicitly don't want to refactor the old idle loop, lest I break it and not notice.